libr/anal/meta.c

```diff
@@ -27,16 +27,18 @@
 #include <r_print.h>
 
 #define META_RANGE_BASE(x) ((x)>>12)
-#define META_RANGE_SIZE 0xfff
 #undef DB
 #define DB a->sdb_meta
 
 static char *meta_inrange_get (RAnal *a, ut64 addr, int size) {
+	if (size <= 0) {
+		return NULL;
+	}
 	ut64 base = META_RANGE_BASE (addr);
-	ut64 base2 = META_RANGE_BASE (addr + size) + 1;
+	ut64 base2 = META_RANGE_BASE (addr + size - 1);
 	char *res = NULL;
 	// return string array of all the offsets where there are stuff
-	for (; base < base2; base += META_RANGE_SIZE) {
+	for (; base <= base2; base++) {
 		const char *key = sdb_fmt ("range.0x%"PFMT64x, base);
```

Comment:
uhmf so the base now is no longer sized in RANGE_SIZE, but 1 by 1?

Reply:
If I understand it correctly, range.0xI is set if there is metadata in [i*4096, (i+1)*4096).

The updated code achives this, before, it was incorrect.